Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -197,6 +200,247 @@
     return m_rootViewToPluginTransform.mapPoint(point);
 }
 
+void PDFPluginBase::updateControlTints(GraphicsContext& graphicsContext)
+{
+    ASSERT(graphicsContext.invalidatingControlTints());
+
+    if (m_horizontalScrollbar)
+        m_horizontalScrollbar->invalidate();
+    if (m_verticalScrollbar)
+        m_verticalScrollbar->invalidate();
+    invalidateScrollCorner(scrollCornerRect());
+}
+
+IntRect PDFPluginBase::scrollCornerRect() const
+{
+    if (!m_horizontalScrollbar || !m_verticalScrollbar)
+        return IntRect();
+    if (m_horizontalScrollbar->isOverlayScrollbar()) {
+        ASSERT(m_verticalScrollbar->isOverlayScrollbar());
+        return IntRect();
+    }
+    return IntRect(m_view->width() - m_verticalScrollbar->width(), m_view->height() - m_horizontalScrollbar->height(), m_verticalScrollbar->width(), m_horizontalScrollbar->height());
+}
+
+ScrollableArea* PDFPluginBase::enclosingScrollableArea() const
+{
+    // FIXME: Walk up the frame tree and look for a scrollable parent frame or RenderLayer.
+    return nullptr;
+}
+
+IntRect PDFPluginBase::scrollableAreaBoundingBox(bool*) const
+{
+    return m_view->frameRect();
+}
+
+void PDFPluginBase::setScrollOffset(const ScrollOffset& offset)
+{
+    m_scrollOffset = IntSize(offset.x(), offset.y());
+
+    didChangeScrollOffset();
+}
+
+bool PDFPluginBase::isActive() const
+{
+    if (auto* coreFrame = m_frame ? m_frame->coreLocalFrame() : nullptr) {
+        if (auto* page = coreFrame->page())
+            return page->focusController().isActive();
+    }
+
+    return false;
+}
+
+bool PDFPluginBase::forceUpdateScrollbarsOnMainThreadForPerformanceTesting() const
+{
+    if (auto* coreFrame = m_frame ? m_frame->coreLocalFrame() : nullptr) {
+        if (auto* page = coreFrame->page())
+            return page->settings().scrollingPerformanceTestingEnabled();
+    }
+
+    return false;
+}
+
+ScrollPosition PDFPluginBase::scrollPosition() const
+{
+    return IntPoint(m_scrollOffset.width(), m_scrollOffset.height());
+}
+
+ScrollPosition PDFPluginBase::minimumScrollPosition() const
+{
+    return IntPoint();
+}
+
+ScrollPosition PDFPluginBase::maximumScrollPosition() const
+{
+    IntSize scrollbarSpace = scrollbarIntrusion();
+    auto pdfDocumentSize = contentsSize();
+
+    IntPoint maximumOffset(pdfDocumentSize.width() - m_size.width() + scrollbarSpace.width(), pdfDocumentSize.height() - m_size.height() + scrollbarSpace.height());
+    maximumOffset.clampNegativeToZero();
+    return maximumOffset;
+}
+
+float PDFPluginBase::deviceScaleFactor() const
+{
+    if (m_frame) {
+        if (auto* coreFrame = m_frame->coreLocalFrame()) {
```

Comment:
should these be Ref / RefPtr instead of raw pointers?

Reply:
It probably doesn't hurt but it is also not strictly required in this case since we're only calling trivial getters.